crates/misc/component-fuzz-util/src/lib.rs

```diff
@@ -76,45 +72,28 @@ impl<'a, const L: usize, const H: usize> Arbitrary<'a> for UsizeInRange<L, H> {
     }
 }
 
-/// Wraps a `Box<[T]>` and provides an `Arbitrary` implementation that always generates non-empty slices
-#[derive(Debug)]
-pub struct NonEmptyArray<T>(Box<[T]>);
-
-impl<'a, T: Arbitrary<'a>> Arbitrary<'a> for NonEmptyArray<T> {
-    fn arbitrary(input: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
-        Ok(Self(
-            iter::once(input.arbitrary())
-                .chain(input.arbitrary_iter()?)
-                .collect::<arbitrary::Result<_>>()?,
-        ))
-    }
-}
-
-impl<T> Deref for NonEmptyArray<T> {
-    type Target = [T];
-
-    fn deref(&self) -> &[T] {
-        self.0.deref()
-    }
-}
-
 /// Wraps a `Box<[T]>` and provides an `Arbitrary` implementation that always generates slices of length less than
 /// or equal to the longest tuple for which Wasmtime generates a `ComponentType` impl
 #[derive(Debug)]
-pub struct TupleArray<T>(Box<[T]>);
+pub struct VecInRange<T, const L: usize, const H: usize>(Vec<T>);
 
-impl<'a, T: Arbitrary<'a>> Arbitrary<'a> for TupleArray<T> {
+impl<'a, T: Arbitrary<'a>, const L: usize, const H: usize> Arbitrary<'a> for VecInRange<T, L, H> {
     fn arbitrary(input: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
-        Ok(Self(
-            input
-                .arbitrary_iter()?
-                .take(MAX_TUPLE_LENGTH)
-                .collect::<arbitrary::Result<_>>()?,
-        ))
+        let mut ret = Vec::new();
+        while ret.len() < L {
+            ret.push(input.arbitrary()?);
+        }
+        for val in input.arbitrary_iter()? {
+            ret.push(val?);
+            if ret.len() == H {
+                break;
+            }
+        }
```

Comment:
I think this is exactly what `Unstructured::arbitrary_loop` is meant for:

```suggestion
        input.arbitrary_loop(Some(L), Some(H), |input| {
            ret.push(input.arbitrary()?);
            Ok(std::ops::ControlFlow::Continue(()))
        });
```